ROADMAP.md

```diff
@@ -0,0 +1,154 @@
+## Vision
+
+> Amethyst aims to be a modular, parallel and data-driven game engine
+
+**Modularity** The game engine should be easily extendable to fit the needs of the game developers for their game idea. Therefore we expose an entity-component system, which allows to define new components and processors without needing to change the game engine code. This should emphasize the 'game engine as library' aspect.
```

Comment:
How opinionated do we want to be for each subsystem? For example, when we integrate scripting, do we want to pick a particular option as default while still leaving the option of switching it out for something else, or do we leave that decision completely up to the user?

I'm in favor of being opinionated, as that makes getting started for new users a lot easier.